maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -196,14 +200,16 @@ public static void main( String[] args )
 
     public static int main( String[] args, ClassWorld classWorld )
     {
-        MavenCli cli = new MavenCli();
-
         MessageUtils.systemInstall();
         MessageUtils.registerShutdownHook();
-        int result = cli.doMain( new CliRequest( args, classWorld ) );
-        MessageUtils.systemUninstall();
-
-        return result;
+        try
+        {
+            return doMain( args, classWorld );
+        }
+        finally
+        {
+            MessageUtils.systemUninstall();
+        }
```

Comment:
Although this changes the behavior, but I think this is good because it is guaranteed that Jansi won't leave a ditry env in case of an expception in `doMain()`.

Reply:
Yes, that's the point.